protobuf/protoc-gen-cruxclient/api_generator.cc

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2020 SafetyCulture Pty Ltd. All Rights Reserved.
+#include "api_generator.h"
+#include <google/protobuf/compiler/code_generator.h>
+#include <google/protobuf/compiler/plugin.h>
+#include <google/protobuf/descriptor.h>
+#include <google/protobuf/io/printer.h>
+#include <google/protobuf/io/zero_copy_stream.h>
+#include <iostream>
+#include <vector>
+#include <map>
+#include <memory>
+#include "common.h"
+
+using google::protobuf::Descriptor;
+using google::protobuf::FileDescriptor;
+using google::protobuf::MethodDescriptor;
+using google::protobuf::ServiceDescriptor;
+using google::protobuf::compiler::CodeGenerator;
+using google::protobuf::compiler::GeneratorContext;
+using google::protobuf::compiler::ParseGeneratorParameter;
+using google::protobuf::compiler::PluginMain;
+using google::protobuf::io::Printer;
+using google::protobuf::io::ZeroCopyOutputStream;
+using std::string;
+
+namespace cruxclient_generator {
+
+void APIGenerator::PrintPrologue(
+  Printer *printer,
+  const FileDescriptor *file) const {
+  std::map<string, string> vars;
+  vars["filename"] = file->name();
+  printer->Print("// Generated by the CRUX Engine C++ plugin.\n");
+  printer->Print("// If you make any local change, they will be lost.\n");
+  printer->Print(vars, "// source: $filename$\n");
+}
+
+void APIGenerator::PrintNamespace(
+  Printer *printer,
+  const FileDescriptor *file,
+  bool isEpilogue) const {
+  std::map<string, string> vars;
+  if (!file->package().empty()) {
+    vars["namespace"] = DotsToColons(file->package());
+
+    if (!isEpilogue) {
+      printer->Print(vars, "namespace $namespace$ {\n");
+    } else {
+      printer->Print(vars, "}  // namespace $namespace$\n");
+    }
+    printer->Print(vars, "\n");
+  }
+}
+
+void APIGenerator::PrintHeaderPrologue(
+  Printer *printer,
+  const FileDescriptor *file) const {
+  std::map<string, string> vars;
+  vars["filename_identifier"] = FilenameIdentifier(file->name());
+
+  PrintPrologue(printer, file);
+}
+
+std::string APIGenerator::GetMethodSignature(
+  const std::string& service_name,
+  const std::string& method_name) const {
+  return "k" + service_name + method_name;
```

Comment:
Never a fan of 1 line functions; can this be in-lined?

Reply:
Actually, I can delete this.

Reply:
Updated